src/mscorlib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -36,57 +36,37 @@ private static EqualityComparer<T> CreateComparer()
             Contract.Ensures(Contract.Result<EqualityComparer<T>>() != null);
             
             object result = null;
-            RuntimeType t = (RuntimeType)typeof(T);
+            var type = (RuntimeType)typeof(T);
             
-            // Specialize type byte for performance reasons
-            if (t == typeof(byte)) {
+            // Specialize for byte so Array.IndexOf is faster.
+            if (type == typeof(byte))
```

Comment:
@jkotas It's going to be kind of convoluted to create `new ObjectComparer<T>()` in that case because then we'd need to use `Activator.CreateInstance`... what about

```cs
static readonly EqualityComparer<T> defaultComparer = (EqualityComparer<T>)ComparerHelpers.CreateComparer(typeof(T)) ?? new ObjectComparer<T>();
```

?